File: jacinta-helm/day-01/tip-calculator.js
function calculateTip(amount, rating) {
    // standardize the rating input
    let lowerCaseRating = rating.toLowerCase();

    if (lowerCaseRating === "terrible") {
        return 0;
    } else if (lowerCaseRating === "poor") {
        return Math.ceil(amount * 0.05);
    } else if (lowerCaseRating === "good") {
        return Math.ceil(amount * 0.10);
    } else if (lowerCaseRating === "great") {
        return Math.ceil(amount * 0.15);
    } else if (lowerCaseRating === "excellent") {
        return Math.ceil(amount * 0.20);
    } else {
        return "Rating not recognised"
    }

}

// Parameter create a function that calculates tip based on bill and service
// Return the value of tip amount
// Example (30, "Good"), 3
// Create a string for the value of ratings
// use if else statements for function
// Concerns rounding up whole numbers

/* Ember's Feedback:
    Very good job!

    Giving my solution below "for the soup"
    Concerns for this problem would be if amount was not a number, or -
    rating not being a string, because we assume it will have toLowerCase().
*/
// Alternative Solution
function calculateTip(amount, rating) {
    if (isNaN(amount)) return 0;

    const tip = (percent) => Math.ceil(amount * percent);

    switch (rating?.toLowerCase()) {
        case "terrible":
            return 0;
        case "poor":
            return tip(0.05);
        case "good":
            return tip(0.1);
        case "great":
            return tip(0.15);
        case "excellent":
            return tip(0.2);
        default:
            return "Rating not recognised";
    }
}

File: eoin-mcdonnell/day-27/tip_calculator.py
# switch case in python
# math ceil in python
import math


def calculate_tip(amount, rating):
    rates = {"terrible": 0, "poor": 5, "good": 10, "great": 15, "excellent": 20}
    if not rating or rating.lower() not in rates:
        return "Rating not recognised"
    else:
        return math.ceil(amount * rates[rating.lower()] / 100)


"""
    Ember's Feedback:
    - Great job! I like that syntax in the if else
"""


# Alternative Solutions
def calculate_tip(amount, rating):
    how_was_service = {
        "terrible": 0.0,
        "poor": 0.05,
        "good": 0.1,
        "great": 0.15,
        "excellent": 0.2,
    }

    rating = rating.lower()

    return (
        math.ceil(amount * how_was_service[rating])
        if rating in how_was_service
        else "Rating not recognised"
    )


def calculate_tip(amount, rating):
    match rating.lower():
        case "terrible":
            mult_value = 0
        case "poor":
            mult_value = 0.05
        case "good":
            mult_value = 0.10
        case "great":
            mult_value = 0.15
        case "excellent":
            mult_value = 0.20
        case _:
            return "Rating not recognised"

    return math.ceil(amount * mult_value)
